AirCasting/Dashboard/DashboardView.swift

```diff
@@ -37,58 +48,97 @@ struct DashboardView: View {
             //
             // Bug report was filled with Apple
             PreventCollapseView()
-            AirSectionPickerView(selection: self.$selectedSection.selectedSection)
-                .padding(.leading)
-                .background(
-                    ZStack(alignment: .bottom) {
-                        Color.green
-                            .frame(height: 3)
-                            .shadow(color: Color.aircastingDarkGray.opacity(0.4),
-                                    radius: 6)
-                            .padding(.horizontal, -30)
-                        Color.white
-                    }
-                )
-                .zIndex(2)
-            if sessions.isEmpty {
-                if selectedSection.selectedSection == .mobileActive || selectedSection.selectedSection == .mobileDormant {
-                    EmptyMobileDashboardViewMobile()
-                } else {
-                    EmptyFixedDashboardView()
-                }
-            } else {
-                ZStack(alignment: .bottomTrailing) {
-                    Image("dashboard-background-thing")
-                    let thresholds = Array(self.thresholds)
-                    ScrollView(.vertical) {
-                        LazyVStack(spacing: 8) {
-                            ForEach(sessions.filter { $0.uuid != "" }, id: \.uuid) { session in
-                                let followingSetter = MeasurementStreamStorageFollowingSettable(session: session, measurementStreamStorage: measurementStreamStorage)
-                                let viewModel = SessionCartViewModel(followingSetter: followingSetter)
-                                SessionCartView(session: session,
-                                                sessionCartViewModel: viewModel,
-                                                thresholds: thresholds,
-                                                sessionStoppableFactory: sessionStoppableFactory,
-                                                measurementStreamStorage: measurementStreamStorage)
-                            }
-                        }
-                    }
-                }
-                .padding(.horizontal)
-                .frame(maxWidth: .infinity)
-                .background(Color.aircastingGray.opacity(0.05))
-            }
+            sessionTypePicker
+            if sessions.isEmpty { emptySessionsView } else { sessionListView }
         }
         .navigationBarTitle(NSLocalizedString(Strings.DashboardView.dashboardText, comment: ""))
         .onChange(of: selectedSection.selectedSection) { selectedSection in
             self.selectedSection.selectedSection = selectedSection
             try! coreDataHook.setup(selectedSection: self.selectedSection.selectedSection)
         }
+        .onChange(of: isRefreshing, perform: { newValue in
+            guard newValue == true else { return }
+            guard !sessionSynchronizer.syncInProgress.value else {
```

Comment:
I think, that maybe when we try to run syncing and there is already sync in progress (like here or on log out), we should check if there were any changes introduced already after the sync started so that we could schedule another sync with them. For example, there is a sync already in progress, but someone decides they want to delete one more session so they delete it and then pull to refresh. Then this deleted session would not be deleted from backed until another sync happens, isn't that right? Do you think that's something you could add in this PR? Maybe first version we could trigger another sync in closure below, but it's probably not an optimal solution.

Reply:
That is an excellent point and I think should be considered more widely since we're also triggering synchronization when user stops a session. So there is additional scenario:
* Sync is running because of coming back from background (for example)
* User stops a sesssion
* It won't get synchronized until next sync
And since we removed the 5-min timer synchronization it won't until user pulls to refresh. This is a problem and it's great you found it so quickly. I will resolve it on a higher level by adding `.required` option or smth like that to the synchronizer.

btw. @anna1901 @anna1901 @mturnau we're not triggering synchronization when deleting sessions, shouldn't we add it too?